docs/_rpc/server.md

```diff
@@ -3,87 +3,181 @@ title: JSON-RPC Server
+If you want to enable access to the API from a web page, 
+you must configure the server to accept Cross-Origin requests 
+with the `--http.corsdomain` flag. Also, Also, you can add more than one domain
+to accept cross-origin requests by separating them with a comma (,).
+
+> you can use `--http.corsdomain '*'` to enable access from any origin, but it is highly insecure and not recommended.
+ 
+ Example: if you want to use [Remix][remix] with geth, it will allow requests from the
 remix domain.
 
     geth --http --http.corsdomain https://remix.ethereum.org
 
-Use `--http.corsdomain '*'` to enable access from any origin.
+
+To enable access to the virtual hostnames from a server, 
+you will configure geth http server to accept requests from 
+that hostname with the `--http.vhosts` flag. Also, you can add 
+more than one hostname by separating them with a comma (,).
+
+    geth --http --http.vhosts http://localhost:9002/
 
 ### WebSocket Server
 
-Configuration of the WebSocket endpoint is similar to the HTTP transport. To
-enable WebSocket access, use `--ws` flag. The default WebSocket port is 8546.
-The `--ws.addr`, `--ws.port` and `--ws.api` flags can be used to customize settings
-for the WebSocket server.
+The configuration of the WebSocket endpoint is similar to the HTTP transport. 
+This `--ws` flag enables the web socket RPC API Server in geth 
+and listens to `ws://127.0.0.1:8546` as the default IP and port.
+
+    geth --ws
+
+
+Geth web socket listens to localhost by default (127.0.0.1).
+However, by modifying the value with this flag `--ws.addr`, 
+you can alter the address that geth listens to via the 
+geth websocket. 
+
+    geth --ws --ws.addr ws://190.0.0.1
+
+
+If you want to enable access to the server from a web page, 
+you must configure the server to accept Cross-Origin requests 
+with the `--ws.origins` flag. Also, you can add more than 
+one domain to accept cross-origin requests by separating them with a comma (,).
+
+> As with `--http.corsdomain`, using `--ws.origins '*'` allows access from any origin.
+
+
+    geth --ws --ws.origins chrome-extension://fgponpodhbmadfljofbimhhlengambbn
+  
+
+Geth web socket listens to port 8546 by default.
+However, by setting the value of the flag `--ws.port`, 
+you can alter the address that geth listens to via the WS-RPC server . 
+    
+     geth --ws --ws.port 8080
+   
+
+The default whitelist allows access to the “eth” and “shh” namespaces. 
+To enable access to other APIs like account management (“personal”) 
+and debugging (“debug”), they must be configured via the `--ws.api` flag.
+However, we do not recommend enabling such APIs over WebSocket since 
+access to these methods increases the attack surface.
 
     geth --ws --ws.port 3334 --ws.api eth,net,web3
 
-Cross-Origin request protection also applies to the WebSocket server. Use the
-`--ws.origins` flag to allow access to the server from web pages:
+### GraphQl
```

Comment:
I had created a GraphQL page not too long ago :) Do you want to either integrate those info here or these ones over there? to avoid duplication